Mathlib/Order/Basic.lean

```diff
@@ -654,6 +654,23 @@ theorem update_le_update_iff {ι : Type u} {α : ι → Type v} [∀ i, Preorder
     Function.update x i a ≤ Function.update y i b ↔ a ≤ b ∧ ∀ (j) (_ : j ≠ i), x j ≤ y j := by
   simp (config := { contextual := true }) [update_le_iff]
 
+
+@[simp] lemma le_update_self_iff {ι : Type u} {α : ι → Type v} [∀ i, Preorder (α i)]
```

Comment:
I've created  #1112 which conflicts with this. Let me know how you'd prefer to proceed.

Reply:
Go ahead with your one, I'll remove the `Mathlib.Order.Basic` stuff here, then.